WalletWasabi.Tests/UnitTests/WabiSabi/Backend/Rounds/MultipartyTransactionStateTests.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using NBitcoin;
+using WalletWasabi.Tests.Helpers;
+using WalletWasabi.WabiSabi.Backend;
+using WalletWasabi.WabiSabi.Models.MultipartyTransaction;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.WabiSabi.Backend.Rounds;
+
+public class MultipartyTransactionStateTests
+{
+	[Fact]
+	public void CanGetDifferentialStateTest()
+	{
+		var cfg = new WabiSabiConfig();
+		var round = WabiSabiFactory.CreateRound(cfg);
+
+		static Coin CreateCoin()
+		{
+			using var key = new Key();
+			return WabiSabiFactory.CreateCoin(key);
+		}
+
+		Coin coin1, coin2, coin3;
+
+		// Three events / three states
+		var state0 = round.Assert<ConstructionState>();
+		var state1 = state0.AddInput(coin1 = CreateCoin());
+		var state2 = state1.AddInput(coin2 = CreateCoin());
+		var state3 = state2.AddInput(coin3 = CreateCoin());
+
+		// Unknown state. Assumes full state is required
+		var diffd30 = state3.GetConstructionStateSince(-1);
```

Comment:
`-1` should be probably a constant because it requires checking that LINQ's `Skip` function returns all elements with negative numbers.

Reply:
What I want to prove here is that when the index requested is out of bound then the full status is returned. I could add an `OutOfBoundIndex` constant but that doesn't add much, the comment is enough i think.